core/src/main/java/org/apache/iceberg/util/TableScanUtil.java

```diff
@@ -48,13 +58,42 @@ public static boolean hasDeletes(FileScanTask task) {
   }
 
   public static CloseableIterable<CombinedScanTask> planTasks(CloseableIterable<FileScanTask> splitFiles,
-                                                              long splitSize, int lookback, long openFileCost) {
+      long splitSize, int lookback, long openFileCost, PartitionSpec spec) {
```

Comment:
I don't think there's meaningful to change this method to accept an extra `PartitionSpec` argument, because we will set the `preservedPartitionIndices`  to be `null` in the real implementation.  Finally,  we won't do any partition GROUP BY in the real path.

I think we could keep the method definition just as it is.